### ISSUE TITLE
Raster clipping

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/geoutils#583]
It would be great to have a function to clip a raster to a vector. Currently, the function `Raster.crop` if passed a vector can crop to the bounds of the vector, but it will not clip (set values to nodata outside the vector outlines). 
`Vector.crop` has an option "clip" to do that. Maybe we could just add a similar "clip" option to `Raster.crop` that works when the input is a vector?

**Comment by rhugonnet** [comment]
Good idea, would basically be the equivalent of `crop` + `create_mask` + `set_mask`?

**Comment by adehecq** [comment]
> Good idea, would basically be the equivalent of `crop` + `create_mask` + `set_mask`?

I think so !
